addons/product/models/product_supplierinfo.py

```diff
@@ -31,14 +23,12 @@ def _default_product_id(self):
     sequence = fields.Integer(
         'Sequence', default=1, help="Assigns the priority to the list of product vendor.")
     product_uom_id = fields.Many2one(
-        'uom.uom', 'Unit of Measure',
-        related='product_tmpl_id.uom_po_id')
+        'uom.uom', 'Unit', compute='_compute_product_uom_id', store=True, readonly=False)
     min_qty = fields.Float(
         'Quantity', default=0.0, required=True, digits="Product Unit of Measure",
         help="The quantity to purchase from this vendor to benefit from the price, expressed in the vendor Product Unit of Measure if not any, in the default unit of measure of the product otherwise.")
```

Comment:
Don't forget to update "Unit of Measure" => "Unit" everywhere possible/noticeable for consistency. I guess in this case, we can also make the help more clear too + match your renaming of price :thinking: 

```suggestion
        'Quantity', default=0.0, required=True, digits="Product Unit of Measure",
        help="The quantity to purchase from this vendor to benefit from the unit price. If a vendor unit is set, quantity should be specified in this unit, otherwise it should be specified in the default unit of the product.")
```